### DISCUSSION TITLE
How to implement "add user to group by scanning QR code"

### DISCUSSION THREAD

**Discussion by JakesMD** [supabase/supabase#26337] [Questions]
Hi!

I need advice on how best to implement this feature:
A group owner can add a user to their group by scanning a QR code on the user's device. A user can only be a member of one group.

Possible implementations I've thought of:
- (easy) QR code is simply the UID of the user. Scanning it will trigger a Postgres function that adds the UID as a member. (is this even secure?)
- (medium) Add a "invite_token" as a custom claim when the user signs up. The QR code is the invite token. The invite token is removed from the custom claim when used.
- (hard) Create separate table for invite tokens. The user requests a new token to display as a QR code. The invite token is removed from the table when used, or expires.

Thanks!

**Comment by roo12312** [comment]
Were you able to solve it? Even i want to authenticate via qr code
